packages/aikit-ui/src/lib/openai-transport.ts

```
@@ -183,7 +189,9 @@ function handleSSEStream(
         ctrl.enqueue({ type: "start", messageId });
       },
       transform(result, ctrl) {
+        if (hasError) return;
```

Is this hiding the actual error message?

---

Before any error `hasError` is false and set to true only once an error happens. This condition has been included to make sure that any residual streaming chunks are not returned to the client in case an error has happened.

---

Why shouldn't we return residual streaming chunks?

---

If you return residual streaming chunks you may update the ui showing the message content _after_ the user has already received an error, making the feedback confusing.

Here I want to make sure the client receives a consistent stream: either successful content OR an error, never a confusing mix.